src/visualizer/detailsElement.js

```diff
@@ -0,0 +1,257 @@
+     * @returns {number} returns the average value of the innertext of the cells with the classname
+     */
+    function sumByTableCellClassName(className){
+        const cells = document.querySelectorAll("." + className);
+        let sum = 0;
+        let groupcounter = 0;
+        for (const cell of cells) {
+            if (cell.innerText.toString().includes("Group-size")){
+                sum += parseInt(cell.innerText.replace("Group-size: ", ""));
+            }
+            else {
+                sum += parseInt(cell.innerText);
+            }
+            groupcounter++;
+        }
+        return (sum / groupcounter).toFixed(2);
+    }
+    /**
+     * @summary Creates the html table header, only the first parameter is used in calls
+     * @param {string} h1 The innertext of the first header cell
+     * @param {string} h2 The innertext of the second header cell
+     * @param {string} h3 The innertext of the third header cell
+     * @param {string} h4 The innertext of the fourth header cell
+     * @returns {HTMLTableRowElement} return a html row element with the table header
+     */
+    function createTableHeader(h1, h2 = "MaxMin", h3 = "Sum", h4 = "LinearDistribution"){
+        const tr = document.createElement("tr");
+        tr.appendChild(createTableTh(h1));
+        tr.appendChild(createTableTh(h2));
+        tr.appendChild(createTableTh(h3));
+        tr.appendChild(createTableTh(h4));
+        return tr;
+    }
+    /**
+     * @summary Creates a html table headercell element
+     * @param {string} text the name of the headercell
+     * @returns {HTMLTableHeaderCellElement} returns a html table header cell element
+     */
+    function createTableTh(text){
+        const th = document.createElement("th");
+        th.innerText = text;
+        return th;
+    }
+    /**
+     * @summary Makes a table cell
+     * @param {string} value The innertext of the cell
+     * @param {string} className The name of the class to be assigned to the cell
+     * @returns {HTMLTableCellElement} returns a html table cell
+     */
+    function createTableTd(value, className){
+        const td = document.createElement("td");
+        if (className !== undefined){
+            td.setAttribute("class", className);
+        }
+        td.innerText = value;
+        return td;
+    }
+    // --------------------------------------Details stuff----------------------------------
+    /**
+     * @summary Creates a list as a html element
+     * @param {object} group Takes a class of groups with all the students in the class
+     * @returns {HTMLElement} Returns a list as a html element
+     */
+    function createGroupElement(group) {
+        const details = createDetailsElement("Groupname: " + group.name, "Group");
+        const studentList = document.createElement("ul");
+        for (let i = 0; i < group.students.length; i++) {
+            studentList.appendChild(createStudentElement(group.students[i], i));
```

Comment:
Hvorfor tager i i ind her?

Reply:
Fordi den studerendes plads i gruppen bliver brugt til at vælge farve

Reply:
Send farven med ind i stedet for

Reply:
Ok